modules/kubeconfig/main.tf

```diff
@@ -37,8 +29,33 @@ data "template_file" "kubeconfig_sa" {
     token        = lookup(data.kubernetes_secret.sa_secret.0.data,"token", "")
     namespace    = var.namespace
   }
+  depends_on = [kubernetes_secret.sa_secret]
+}
+
+# 1.24 change: Create service account secret
+resource "kubernetes_secret" "sa_secret" {
+  count = var.create_static_kubeconfig ? 1 : 0
+  metadata {
+    name      = local.service_account_secret_name
+    namespace = var.namespace
+    annotations = {
+      "kubernetes.io/service-account.name" = local.service_account_name
+    }
+  }
+  type = "kubernetes.io/service-account-token"
+  depends_on = [kubernetes_service_account.kubernetes_sa]
+}
+
+data "kubernetes_secret" "sa_secret" {
```

Comment:
Also, I would suggest to add this block back up to line #20 as it was originally.

Reply:
Latest change moved it back